Dockerfile

```diff
@@ -0,0 +1,71 @@
+# Example build command:
+# export FROM_IMAGE="ros:foxy"
+# export OVERLAY_MIXINS="release ccache"
+# docker build -t chapulina/dolly:foxy \
+#   --build-arg FROM_IMAGE \
+#   --build-arg OVERLAY_MIXINS \
+#   -f Dockerfile ./
+ARG FROM_IMAGE=ros:foxy
+ARG OVERLAY_WS=/opt/overlay_ws
+
+# multi-stage for caching
+FROM $FROM_IMAGE AS cacher
+
+# clone underlay source
+ARG OVERLAY_WS
+WORKDIR $OVERLAY_WS/src
+COPY ./ ./chapulina/dolly
+
+# copy manifests for caching
+WORKDIR /opt
+RUN mkdir -p /tmp/opt && \
+    find ./ -name "package.xml" | \
+      xargs cp --parents -t /tmp/opt && \
+    find ./ -name "COLCON_IGNORE" | \
+      xargs cp --parents -t /tmp/opt || true
+
+# multi-stage for building
+FROM $FROM_IMAGE AS builder
+ARG DEBIAN_FRONTEND=noninteractive
+
+# install CI dependencies
+RUN apt-get update && apt-get install -q -y \
+        ccache \
+        lcov \
+        ros-${ROS_DISTRO}-ros-ign \
```

Comment:
Shouldn't this dependency be explicitly declared in a package manifest?

Reply:
I reverted 52e55a9 and rebased from #26 , but I still see this cmake warning that skips the build:

```
[1.843s] WARNING:colcon.colcon_cmake.task.cmake.build:Could not run installation step for package 'dolly_ignition' because it has no 'install' target
--- stderr: dolly_ignition
CMake Warning at CMakeLists.txt:8 (message):
  Ignition Gazebo 3 not found, proceeding without that simulator.
```

Rebuild using b41f4d5

```
docker build -t chapulina/dolly:foxy --progress plain .
```

* Are you sure your using the right rosdep key for the package depend?
  * i.e. `ros_ign` rather than `ros_ign_gazebo`, or `*-dev` needed?
* is there an issue with the released ignition metapackages?
  * is an implicit relation missing or link to `*-dev` package?

Reply:
Ok, I think there are more problems upstream: https://github.com/ignitionrobotics/ros_ign/pull/110